File: publisher-service/src/main/java/com/souqstation/publisher/messaging/PlatformFeedbackConsumer.java
package com.souqstation.publisher.messaging;

import org.apache.avro.generic.GenericRecord;
import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.kafka.support.Acknowledgment;
import org.springframework.stereotype.Component;

@Component
public class PlatformFeedbackConsumer {

    @KafkaListener(topics = "${souq.topics.platform}")
    public void onPlatformEvent(ConsumerRecord<String, GenericRecord> record, Acknowledgment ack) {
        GenericRecord payload = record.value();
        if (payload == null) {
            ack.acknowledge();
            return;
        }

        String schemaName = payload.getSchema().getName();
        if (!"ReviewSubmittedEvent".equals(schemaName) && !"IncidentReportedEvent".equals(schemaName)) {
            ack.acknowledge();
            return;
        }

        Object eventId = payload.get("eventId");
        Object gameId = payload.get("gameId");

        System.out.println("[PUBLISHER] feedback consumed"
                + " key=" + record.key()
                + " schema=" + schemaName
                + " eventId=" + eventId
                + " gameId=" + gameId);

        ack.acknowledge();
    }
}

File: publisher-service/src/main/java/com/souqstation/publisher/messaging/PlatformFeedbackConsumer.kt
package com.souqstation.publisher.messaging

import org.apache.avro.generic.GenericRecord
import org.apache.kafka.clients.consumer.ConsumerRecord
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.kafka.support.Acknowledgment
import org.springframework.stereotype.Component

@Component
class PlatformFeedbackConsumer {

    @KafkaListener(topics = ["\${souq.topics.platform}"])
    fun onPlatformEvent(
        record: ConsumerRecord<String, GenericRecord>,
        ack: Acknowledgment
    ) {
        val payload = record.value()
        if (payload == null) {
            ack.acknowledge()
            return
        }

        val schemaName = payload.schema.name
        if (schemaName != "ReviewSubmittedEvent" &&
            schemaName != "IncidentReportedEvent"
        ) {
            ack.acknowledge()
            return
        }

        val eventId = payload["eventId"]
        val gameId = payload["gameId"]

        println(
            "[PUBLISHER] feedback consumed" +
                    " key=${record.key()}" +
                    " schema=$schemaName" +
                    " eventId=$eventId" +
                    " gameId=$gameId"
        )

        ack.acknowledge()
    }
}